can/interfaces/slcan.py

```diff
@@ -155,21 +155,17 @@ def _read(self, timeout: Optional[float]) -> Optional[str]:
 
         with error_check("Could not read from serial device"):
             while True:
-                new_data = self.serialPortOrig.read()
-                if new_data:
-                    self._buffer.extend(new_data)
-                else:
-                    if _timeout.expired():
-                        break
-                    else:
-                        continue
-
-                for terminator in (self._ERROR, self._OK):
-                    if terminator in self._buffer:
-                        i = self._buffer.index(terminator) + 1
-                        string = self._buffer[:i].decode()
-                        del self._buffer[:i]
-                        return string
+                new_byte = self.serialPortOrig.read(size=1)
+                if new_byte:
+                    self._buffer.extend(new_byte)
+
+                if new_byte in (self._ERROR, self._OK):
```

Comment:
Since `serialPortOrig.read` only reads one byte, the SLCAN message contains multiple bytes.

It may return `None` before receiving a full SLCAN message.